### PR TITLE
Add Pascal datatypes

### PR DESCRIPTION
**Pull request by siddhpant**
Added Pascal data types.

### CODE REVIEW

File: data/DataType/char_pascal.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+keyword: char
+integer-min: '#0'
```
Comment (Naveenaidu) [inline data/DataType/char_pascal.yaml]: Please remove the `#` from here. It is not required.

File: data/Language/Pascal.yaml [hunk 2]
```diff
@@ -72,8 +72,8 @@ keywords:
...
-  - string
-  - boolean
-  - integer
-  - real
-  - char
+  - boolean_pascal
+  - char_pascal
+  - integer_pascal
+  - real_pascal
+  - string_pascal   #TODO
```
Comment (jayvdb) [inline data/Language/Pascal.yaml]: missing starting space in comment (comments)

*Origin: YAMLLintBear, Section: `yaml`.*

**Comment by Naveenaidu** [comment]
Please make the changes, only related to the task. It would be better if you could create a issue later and submit your extra changes via a PR then.

File: data/DataType/char_pascal.yaml (hunk 1 above)
Comment (siddhpant) [reply data/DataType/char_pascal.yaml]: https://www.freepascal.org/docs-html/ref/refsu6.html#x29-320003.2.1 says the char should start with # when ascii code is used....

File: data/Language/Pascal.yaml (hunk 2 above)
Comment (siddhpant) [reply data/Language/Pascal.yaml]: ok

**Comment by siddhpant** [comment]
Okay

File: data/DataType/boolean_title_case.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+name: boolean
+integer-min: '0'
+integer-max: '1'
```
Comment (li-boxuan) [inline data/DataType/boolean_title_case.yaml]: Why `integer-max` and `integer-min` for a boolean variable?
You cannot assign 1 to a boolean variable in Pascal.

File: data/Language/Pascal.yaml [hunk 4]
```diff
@@ -72,8 +72,8 @@ keywords:
...
-  - string
-  - boolean
-  - integer
-  - real
-  - char
+  - boolean_title_case
+  - char_pascal
+  - integer_pascal
+  - real_pascal
+  - string_pascal   # TODO
```
Comment (li-boxuan) [inline data/Language/Pascal.yaml]: I also checked http://wiki.freepascal.org/Variables_and_Data_Types, ya, agree your point that 'Standard Pascal does not make provision for the string data type'. However, it also says 'most modern compilers do'. It's good to keep the string type.
Anyway, leaving a `TODO` here is not very useful 😛

File: data/DataType/boolean_title_case.yaml [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+literals:
+  - 'True'
+  - 'False'
```
Comment (ayan-b) [inline data/DataType/boolean_title_case.yaml]: Please change this to `TRUE` and `FALSE`. ([ref](https://ideone.com/Taf3Lr)) Create a new file named `bool_upper_case.yaml` (if not present already) and put those there.

File: data/DataType/boolean_title_case.yaml (hunk 3 above)
Comment (siddhpant) [reply data/DataType/boolean_title_case.yaml]: ah ok, I originally didn't.

File: data/DataType/boolean_title_case.yaml (hunk 5 above)
Comment (siddhpant) [reply data/DataType/boolean_title_case.yaml]: :+1:

Comment (siddhpant) [reply data/DataType/boolean_title_case.yaml]: Okay

File: data/DataType/string.yaml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+keyword: string
```
Comment (jayvdb) [inline data/DataType/string.yaml]: there is no single most common 'string' , so this file should be called 'string_pascal.yaml' .  Worth nothing that string isnt even a basic data type of the Pascal language.  There are various compilers providing different definitions of what a string is for Pascal and its many derivatives.

Comment (siddhpant) [reply data/DataType/string.yaml]: So should I remove it? 

I added it after this review comment: https://github.com/coala/coAST/pull/82#discussion_r238638338

Comment (siddhpant) [reply data/DataType/string.yaml]: Renamed

File: data/Language/Pascal.yaml [hunk 7]
```diff
@@ -72,8 +72,8 @@ keywords:
...
-  - string
-  - boolean
-  - integer
-  - real
-  - char
+  - boolean_upper_case
+  - char_pascal
+  - integer_pascal
+  - real_pascal
+  - string_pascal
```
Comment (jayvdb) [inline data/Language/Pascal.yaml]: no new line character at the end of file (new-line-at-end-of-file)

*Origin: YAMLLintBear, Section: `yaml`.*

Comment (siddhpant) [reply data/Language/Pascal.yaml]: Hmm

Comment (jayvdb) [inline data/Language/Pascal.yaml]: no new line character at the end of file (new-line-at-end-of-file)

*Origin: YAMLLintBear, Section: `yaml`.*

Comment (siddhpant) [reply data/Language/Pascal.yaml]: Added

**Comment by jayvdb** [comment]
@gitmate-bot rebase

**Comment by jayvdb** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by jayvdb** [comment]
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

**Comment by jayvdb** [comment]
ack https://github.com/coala/coAST/pull/82/commits/d18172fd6700f6ec2df27cc66f7be5d2852cde9a

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Comment by jayvdb** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being fastforwarded automatically. Please **DO NOT** push while fastforward is in progress or your changes would be lost permanently :warning:

**Comment by jayvdb** [comment]
Automated fastforward with [GitMate.io](https://gitmate.io) was successful! :tada:

**Pull request merged**